mix.exs

```diff
@@ -24,20 +24,42 @@ defmodule PublicSuffix.Mixfile do
     ]
   end
 
-  # Dependencies can be Hex packages:
-  #
-  #   {:mydep, "~> 0.3.0"}
-  #
-  # Or git/path repositories:
-  #
-  #   {:mydep, git: "https://github.com/elixir-lang/mydep.git", tag: "1.1.0"}
-  #
-  # Type "mix help deps" for more examples and options
   defp deps do
     [
-      {:idna, "~> 2.0"},
+      {:idna, ">= 1.2.0 and < 3.0.0"},
+      # ex_doc and earmark are necessary to publish docs to hexdocs.pm.
+      {:ex_doc, ">= 0.0.0", only: :dev},
+      {:earmark, ">= 0.0.0", only: :dev},
     ]
   end
+
+  defp description do
+    """
+    Operate on domain names using the public suffix rules provided by https://publicsuffix.org/.
+    """
+  end
```

Comment:
The mix convention is to define it as a private function.  I got this directly from the publishing guide:

https://hex.pm/docs/publish

Consider also that in `mix.exs` you usually define `defp deps`, `defp application`, etc, when those things could be module attributes.